geopandas/io/tests/test_sql.py

```diff
@@ -181,3 +271,333 @@ def test_read_postgis_binary(self, connection_spatialite, df_nybb):
         )
         df = read_postgis(sql, con, geom_col=geom_col)
         validate_boro_df(df)
+
+    def test_write_postgis_default(self, df_nybb):
+        """Tests that GeoDataFrame can be written to PostGIS with defaults."""
+        engine = connect_engine("test_geopandas")
+        if engine is None:
+            raise pytest.skip()
+
+        table = "nybb"
+
+        # If table exists, delete it before trying to write with defaults
+        drop_table_if_exists(engine, table)
+
+        try:
+            # Write to db
+            write_postgis(df_nybb, con=engine, name=table, if_exists="fail")
+            # Validate
+            sql = "SELECT * FROM {table};".format(table=table)
+            df = read_postgis(sql, engine, geom_col="geometry")
+            validate_boro_df(df)
```

Comment:
For some reason, this test is generating those warnings:

```
geopandas/io/tests/test_sql.py::TestIO::test_write_postgis_default
  /home/joris/miniconda3/envs/geo-dev/lib/python3.8/site-packages/sqlalchemy/dialects/postgresql/base.py:3027: SAWarning: Did not recognize type 'geometry' of column 'the_geom'
    util.warn(

geopandas/io/tests/test_sql.py::TestIO::test_write_postgis_default
  /home/joris/miniconda3/envs/geo-dev/lib/python3.8/site-packages/sqlalchemy/dialects/postgresql/base.py:3027: SAWarning: Did not recognize type 'geometry' of column 'geom'
    util.warn(

geopandas/io/tests/test_sql.py::TestIO::test_write_postgis_default
  /home/joris/miniconda3/envs/geo-dev/lib/python3.8/site-packages/sqlalchemy/dialects/postgresql/base.py:3027: SAWarning: Did not recognize type 'geometry' of column 'geometry'
    util.warn(
```

while the other tests are not .. But I don't really understand where this could be coming from

Reply:
Yeah, I have been wondering about the same warnings as well, and don't know either what could be the reason for them.